### PR TITLE
Use new importlib API

### LINKED ISSUE #2934: importlib-metadata DeprecationWarning
Seen on https://github.com/adamchainz/mariadb-dyncol/pull/298 

```
   File "/home/runner/work/mariadb-dyncol/mariadb-dyncol/.tox/py37/lib/python3.7/site-packages/_pytest/assertion/rewrite.py", line 170, in exec_module
    exec(co, module.__dict__)
  File "/home/runner/work/mariadb-dyncol/mariadb-dyncol/.tox/py37/lib/python3.7/site-packages/hypothesis/__init__.py", line 61, in <module>
    run()
  File "/home/runner/work/mariadb-dyncol/mariadb-dyncol/.tox/py37/lib/python3.7/site-packages/hypothesis/entry_points.py", line 61, in run
    for entry in get_entry_points():  # pragma: no cover
  File "/home/runner/work/mariadb-dyncol/mariadb-dyncol/.tox/py37/lib/python3.7/site-packages/hypothesis/entry_points.py", line 32, in get_entry_points
    yield from importlib_metadata.entry_points().get("hypothesis", [])
  File "/home/runner/work/mariadb-dyncol/mariadb-dyncol/.tox/py37/lib/python3.7/site-packages/importlib_metadata/__init__.py", line 312, in get
    flake8_bypass(self._warn)()
DeprecationWarning: SelectableGroups dict interface is deprecated. Use select.
ERROR: InvocationError for command /home/runner/work/mariadb-dyncol/mariadb-dyncol/.tox/py37/bin/python -W error::DeprecationWarning -W error::PendingDeprecationWarning -m pytest (exited with code 1)
```

### PR DESCRIPTION
**Pull request by Zac-HD**
Fixes #2934.

### CODE REVIEW

**Comment by Zac-HD** [comment]
@adamchainz - can you confirm that I'm using the new API correctly?

(Hypothesis' strictly-minimalist approach to mandatory dependencies can be painful sometimes, but it's the only sensible way to support all the various Python implementations we want to run on, including development versions or environments without easy access to PyPI)

File: hypothesis-python/src/hypothesis/entry_points.py [hunk 1]
```diff
@@ -29,7 +29,14 @@
         import importlib_metadata  # type: ignore  # mypy thinks this is a redefinition
 
     def get_entry_points():
-        yield from importlib_metadata.entry_points().get("hypothesis", [])
+        try:
+            eps = importlib_metadata.entry_points(group="hypothesis")
```
Comment (adamchainz) [inline hypothesis-python/src/hypothesis/entry_points.py:33]: I think you want:

`entry_points().select(group="hypothesis")`

**Review by adamchainz — APPROVED** [review]

**Review by adamchainz — CHANGES_REQUESTED** [review]
as above

File: hypothesis-python/src/hypothesis/entry_points.py (hunk 1 above)
Comment (Zac-HD) [reply hypothesis-python/src/hypothesis/entry_points.py:33]: I'm following https://importlib-metadata.readthedocs.io/en/latest/using.html#entry-points here, which says they're equivalent 🤷‍♂️

Comment (adamchainz) [reply hypothesis-python/src/hypothesis/entry_points.py:33]: Ah great, my misunderstanding - I only recall reading the changelog note about this.

**Review by adamchainz — APPROVED** [review]

**Comment by Zac-HD** [comment]
Thanks @adamchainz 😁

**Pull request merged**